core/src/core_tests/queries.rs

```diff
@@ -127,10 +131,10 @@ async fn legacy_query(#[case] include_history: bool) {
 
     let task = worker.poll_workflow_activation().await.unwrap();
     assert_matches!(
-        task.jobs.as_slice(),
-        [WorkflowActivationJob {
+        task.jobs.as_slice()[0],
```

Comment:
Why only check the first element?

Reply:
This can be reverted. I had an earlier change that more eagerly merged evictions but it just caused way too many test failures for me to be comfortable with it.